### PR TITLE
[UT] improve test for kubeconfig.go

### PR DESCRIPTION
**Pull request by my-git9**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
[UT] improve test for kubeconfig.go
**Which issue(s) this PR fixes**:
Ref: https://github.com/karmada-io/karmada/pull/3148#discussion_r1105203347

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/karmadactl/cmdinit/utils/kubeconfig_test.go [hunk 1]
```diff
@@ -133,7 +133,7 @@ func TestWriteKubeConfigFromSpec(t *testing.T) {
 				serverURL:      "https://127.0.0.1:6443",
 				userName:       "admin",
 				clusterName:    "local",
-				kubeconfigPath: "/tmp",
+				kubeconfigPath: "temp-kubeconfig-0214",
```
Comment (XiShanYongYe-Chang) [inline pkg/karmadactl/cmdinit/utils/kubeconfig_test.go:136]: How about using a random suffix?

**Comment by XiShanYongYe-Chang** [comment]
Hi @my-git9, a lint error occurs.

**Comment by XiShanYongYe-Chang** [comment]
thanks~
/lgtm

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**